dbscan/dbscan.go

```diff
@@ -54,13 +55,16 @@ func SnakeCaseMapper(str string) string {
 
 // API is the core type in dbscan. It implements all the logic and exposes functionality available in the package.
 // With API type users can create a custom API instance and override default settings hence configure dbscan.
+// API should not be copied after first use.
 type API struct {
 	structTagKey          string
 	columnSeparator       string
 	fieldMapperFn         NameMapperFunc
 	scannableTypesOption  []interface{}
 	scannableTypesReflect []reflect.Type
 	allowUnknownColumns   bool
+	// columnToIndexFieldMapCache stores a map of reflect.Type -> map[string][]int
+	columnToIndexFieldMapCache sync.Map
```

Comment:
I went back and forth on whether this should store a sync.Map itself or a pointer to a (heap-allocated) one. IMO both ways have risk of improper use. If we store the Map directly, the user could copy the API while the map is being updated. On the other hand, if we use a pointer, the value might not get initialized. (Or we would need to initialize it lazily, which adds some complexity and synchronization overhead.) I'm happy to change to another solution if you feel strongly.